includes/fbbot/class-omise-fbbot-request-handler.php

```diff
@@ -0,0 +1,157 @@
+			$order_id = $checking_text[1];
+			$order_status_message = Omise_FBBot_Conversation_Generator::get_ordet_status_message( $order_id );
+			$response = Omise_FBBot_HTTPService::send_message_to( $sender_id, $order_status_message );
+
+		} else {
+			// Handle unrecognize message
+			$unrecognize_message = Omise_FBBot_Conversation_Generator::unrecognized_message();
+			$response = Omise_FBBot_HTTPService::send_message_to( $sender_id, $unrecognize_message );
+		}
+	}
+
+	public static function handle_payload_from( $sender_id, $payload ) {
+
+		switch ( $payload ) {
+			case Omise_FBBot_Payload::GET_START_CLICKED:
+				$message = Omise_FBBot_Conversation_Generator::greeting_message( $sender_id );
+				$response = Omise_FBBot_HTTPService::send_message_to( $sender_id, $message );
+				break;
+
+			case Omise_FBBot_Payload::FEATURE_PRODUCTS:
+				$message = Omise_FBBot_Conversation_Generator::feature_products_message( $sender_id );
+				$response = Omise_FBBot_HTTPService::send_message_to( $sender_id, $message );
+				break;
+
+			case Omise_FBBot_Payload::PRODUCT_CATEGORY:
+				$message = Omise_FBBot_Conversation_Generator::product_category_message( $sender_id );
+				$response = Omise_FBBot_HTTPService::send_message_to( $sender_id, $message );
+				break;
+
+			case Omise_FBBot_Payload::CHECK_ORDER:
+				$message = Omise_FBBot_Conversation_Generator::before_checking_order_message( $sender_id );
+				$response = Omise_FBBot_HTTPService::send_message_to( $sender_id, $message );
+				break;
+
+			case Omise_FBBot_Payload::HELP:
+				$helping_message = Omise_FBBot_Conversation_Generator::helping_message( $sender_id );
+				$response = Omise_FBBot_HTTPService::send_message_to( $sender_id, $helping_message );
+				break;
+
+			default:
+				# Custom payload :
+				self::handle_custom_payload( $sender_id, $payload ); 
+				break;
+		}
+	}
+
+	private static function handle_custom_payload( $sender_id, $payload ) {
+
+		$explode = explode('__', $payload);
+
+		if ($explode[0] == 'VIEW_PRODUCT') {
+			$product_id = $explode[1];
+			$product_gallery_message = Omise_FBBot_Conversation_Generator::product_gallery_message( $sender_id, $product_id );
+
+			$response = Omise_FBBot_HTTPService::send_message_to( $sender_id, $product_gallery_message );
+
+		} else if ($explode[0] == 'VIEW_CATEGORY_PRODUCTS') {
+			$category_slug = $explode[1];
+			$products_list_message = Omise_FBBot_Conversation_Generator::product_list_in_category_message( $sender_id, $category_slug );
+
+			$result = Omise_FBBot_HTTPService::send_message_to( $sender_id, $products_list_message );
+		} else {
+			// unexpected payload
+		}
+	}
+
+	public static function handle_callback_omise_webhook( $request ) {
+		$body = json_decode( $request->get_body() );
+
+		if ( ( ! isset( $body ) ) || ( ! isset( $body->data ) ) ) 
+			return;
+
+		$charge = $body->data;
+		$metadata = $charge->metadata;
+	
+		if ( ! isset( $metadata ) || ! isset( $metadata->source ) ||  $metadata->source != 'woo_omise_bot' ) {
```

Comment:
@igroomgrim I think here we should check if `$body->data->object == 'charge'` first.